packages/react-core/src/components/Menu/MenuItem.tsx

```diff
@@ -309,6 +309,23 @@ const MenuItemBase: React.FunctionComponent<MenuItemProps> = ({
       setFlyoutRef(null);
     }
   };
+
+  React.useEffect(() => {
+    if (isFocused && ref.current) {
+      const itemEl = ref.current;
+      const parentListEl = itemEl.parentElement;
+
+      if (parentListEl) {
+        const isAboveTop = itemEl.offsetTop - parentListEl.offsetTop < parentListEl.scrollTop;
+        const isBelowBottom = itemEl.offsetTop - parentListEl.offsetTop + itemEl.clientHeight;
+
+        if (isAboveTop || isBelowBottom) {
+          itemEl.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
```

Comment:
"auto" might be a better option for the behavior (unless we'd want to make this customizable/check if a user has some setting that would trigger the `prefers-reduced-motion` media query).

When testing the Select typeahead and passing `isScrollable`, this is the result when traversing through via keyboard then closing/opening after giving focus to the last option:

https://github.com/patternfly/patternfly-react/assets/70952936/8d005b61-6157-472d-993d-6678916b9c95